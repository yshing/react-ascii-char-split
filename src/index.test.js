import React from 'react';
import renderer from 'react-test-renderer';
import AsciiTextSplit from './index.js';

it('Renders some Mixed Text without crashing', () => {
    const div = document.createElement('div');
    const rendered = renderer.create(<AsciiTextSplit>圓周率是3.1415926</AsciiTextSplit>);
    const renderedJSON = rendered.toJSON();
    expect(renderedJSON.children.length).toBe(2);
    expect(renderedJSON.children[0]).toHaveProperty('props.className', 'non-ascii');
    expect(renderedJSON.children[1]).toHaveProperty('props.className', 'ascii');
    expect(renderedJSON).toMatchSnapshot();
});
it('Renders empty text without crashing', () => {
    const div = document.createElement('div');
    const rendered = renderer.create(<AsciiTextSplit />);
    const renderedJSON = rendered.toJSON();
    expect(renderedJSON.children).toBeNull();
    expect(renderedJSON).toMatchSnapshot();
});
it('Renders ascii only text without crashing', () => {
    const div = document.createElement('div');
    const rendered = renderer.create(<AsciiTextSplit>ASCII</AsciiTextSplit>);
    const renderedJSON = rendered.toJSON();
    expect(renderedJSON).toHaveProperty('children.length', 1);
    expect(renderedJSON).toMatchSnapshot();
});
it('Renders alternative class name without issue', () => {
    const div = document.createElement('div');
    const rendered = renderer.create(
        <AsciiTextSplit wrapperClassName="SOMETHING" asciiClassName="number" nonAsciiClassName="chinese">
            圓周率是3.1415926
        </AsciiTextSplit>
    );
    const renderedJSON = rendered.toJSON();
    expect(renderedJSON).toHaveProperty('props.className', 'SOMETHING');
    expect(renderedJSON.children.length).toBe(2);
    expect(renderedJSON.children[0]).toHaveProperty('props.className', 'chinese');
    expect(renderedJSON.children[1]).toHaveProperty('props.className', 'number');
    expect(renderedJSON).toMatchSnapshot();
});
