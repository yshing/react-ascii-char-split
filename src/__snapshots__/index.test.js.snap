// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders alternative class name without issue 1`] = `
<span
  className="SOMETHING"
  style={undefined}
>
  <span
    className="chinese"
    style={undefined}
  >
    圓周率是
  </span>
  <span
    className="number"
    style={undefined}
  >
    3.1415926
  </span>
</span>
`;

exports[`Renders ascii only text without crashing 1`] = `
<span
  className="char-split-wrapper"
  style={undefined}
>
  <span
    className="ascii"
  >
    ASCII
  </span>
</span>
`;

exports[`Renders empty text without crashing 1`] = `<span />`;

exports[`Renders some Mixed Text without crashing 1`] = `
<span
  className="char-split-wrapper"
  style={undefined}
>
  <span
    className="non-ascii"
    style={undefined}
  >
    圓周率是
  </span>
  <span
    className="ascii"
    style={undefined}
  >
    3.1415926
  </span>
</span>
`;
